pennylane_sf/hw.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Strawberry Fields Hardware device
+=================================
+
+**Module name:** :mod:`pennylane_sf.hw`
+
+.. currentmodule:: pennylane_sf.hw
+
+The Strawberry Fields Hardware plugin implements all the :class:`~pennylane.device.Device` methods,
+and provides access to Xanadu's continuous-variable quantum hardware.
+
+Classes
+-------
+
+.. autosummary::
+   StrawberryFieldsRemote
+
+
+Code details
+~~~~~~~~~~~~
+"""
+
+from collections import OrderedDict
+
+import numpy as np
+
+import strawberryfields as sf
+
+# import gates
+from strawberryfields.ops import BSgate, Interferometer, MeasureFock, Rgate, S2gate
+from strawberryfields.utils.post_processing import (
+    all_fock_probs_pnr,
+    samples_expectation,
+    samples_variance,
+)
+
+from .simulator import StrawberryFieldsSimulator
+
+
+class StrawberryFieldsRemote(StrawberryFieldsSimulator):
+    r"""StrawberryFields remote device for PennyLane.
+
+    A valid Strawberry Fields API token is needed for access. This token can be
+    passed when creating the device. The default configuration options of
+    Strawberry Fields are used to store the authentication token in a
+    configuration file.
+
+    Args:
+        wires (int): the number of modes to initialize the device in
+        shots (int): number of circuit evaluations/random samples used to
+            estimate expectation values of observables
+        chip (str): name of the remote chip to be used
+        hbar (float): the convention chosen in the canonical commutation
+            relation :math:`[x, p] = i \hbar`
+        sf_token (str): the SF API token used for remote access
+    """
+    name = "Strawberry Fields Hardware PennyLane plugin"
+    short_name = "strawberryfields.ai"
+
+    _operation_map = {
+        "Beamsplitter": BSgate,
+        "Rotation": Rgate,
+        "TwoModeSqueezing": S2gate,
+        "Interferometer": Interferometer,
+    }
+
+    _observable_map = {
+        "Identity": None,
+        "TensorN": None,
+    }
+
+    def __init__(self, wires, *, chip="X8", shots=1000, hbar=2, sf_token=None):
```

Comment:
Instead of `chip`, I would name it `backend` (for online/remote simulators). Also, best not to provide a default --- it should likely be a required argument.

Reply:
Good point! Changed it